### DISCUSSION TITLE
Welcome to the PhotoDemon Forum!

### DISCUSSION THREAD

**Discussion by tannerhelland** [tannerhelland/PhotoDemon#344] [General]
## 👋 Welcome to the PhotoDemon Forum!

Feel free to use this forum for:
  * Asking questions about PhotoDemon
  * Sharing how-to's and tutorials 
  * Seeking feedback on images you've created or edited with PhotoDemon 

For bug-reports or formal feature requests, [please submit an Issue instead](https://github.com/tannerhelland/PhotoDemon/issues).

Finally, please observe [PhotoDemon's Code of Conduct](https://github.com/tannerhelland/PhotoDemon/blob/84f84be77b7a1f52cb1151eeef8e5df1bbec5fad/CODE_OF_CONDUCT.md) during all forum interactions.

Thank you, and welcome to the community!

**Comment by ericrberry** [comment]
Tanner,  I am excited to use Photodemon.  How do I create a text box and/or multiple text boxes for an image?  Any insights or if you can point me in the right direction I am appreciative.  Thanks

**Reply by tannerhelland** [reply]
Hi @ericrberry !  In PhotoDemon, text is added via **text layers**.  An image can have as many text layers as you want.

Click either of the two T icons on the left toolbar to select a text tool.  (The left one is for **basic text**, the right one is for **advanced text**.)

![image](https://user-images.githubusercontent.com/1930029/108140348-25aa3000-707f-11eb-94c0-15b0855637a7.png)

Once a text tool is active, click-drag on the main canvas to create a new text layer and establish the layer's boundaries.  (You can freely modify boundaries of the layer after it is created, so don't worry about getting it perfect on the first try.)

Once the layer is created, you can set its text, font, and other properties from the toolbar at the bottom of the screen.

...and that's about it!  Please let me know if you run into any trouble.

**Comment by ericrberry** [comment]
Tanner,

Thank you good Sir 👍😊! The timely response is appreciated. Also too is there a limit- character count- for any of the metadata fields?

Sent from ProtonMail mobile

-------- Original Message --------

**Reply by tannerhelland** [reply]
Happy to help!

> is there a limit- character count- for any of the metadata fields?

Yes, some fields do have limits, but they vary widely by file and metadata type.

PhotoDemon wraps the wonderful [ExifTool library](https://exiftool.org/) for all metadata tasks.  The maintainers of ExifTool maintain a comprehensive list of tag types and their various restrictions:

https://exiftool.org/TagNames/index.html

It's a daunting list, which is exactly why PhotoDemon uses their metadata features instead of me trying to write my own.  😉

**Comment by mfmdawdeh** [comment]
Welcome
I am the translator MFM Dawdeh
I would like to translate this program of yours into Arabic.
The program has been translated into Arabic, and the initial translation rate is approximately 90%.
Please kindly reply via email to confirm receipt of this message.

**Reply by tannerhelland** [reply]
Hello @mfmdawdeh !  I apologize for the delay in replying.  I have been on holiday for the past two weeks.

I am very excited to add your Arabic translation to PhotoDemon.  If you can attach your XML translation file to this discussion (or create a new issue with it), that would be perfect.  (We may have to work through some other UI issues in Arabic due to its RTL nature, so I would greatly appreciate any follow-up help you can provide.)

Thank you in advance 🙏

**Comment by mfmdawdeh** [comment]
Welcome
do not worry
Here is the translation
After adding the Arabic language, send me a link to download the program to confirm the translation status.
The translation may need to be modified.
I will tell you if you watch the program and it is in Arabic

________________________________
من: Tanner ***@***.***>
‏‏تم الإرسال: 27 أيلول, 2022 06:58 م
إلى: tannerhelland/PhotoDemon ***@***.***>
نسخة: mfmdawdeh ***@***.***>; Mention ***@***.***>
‏‏الموضوع: Re: [tannerhelland/PhotoDemon] Welcome to the PhotoDemon Forum! (Discussion #344)


Hello @mfmdawdeh<https://github.com/mfmdawdeh> ! I apologize for the delay in replying. I have been on holiday for the past two weeks.

I am very excited to add your Arabic translation to PhotoDemon. If you can attach your XML translation file to this discussion (or create a new issue with it), that would be perfect. (We may have to work through some other UI issues in Arabic due to its RTL nature, so I would greatly appreciate any follow-up help you can provide.)

Thank you in advance 🙏

**Comment by zirrania** [comment]
Hello,
I'm new to the program and just learning how to use it. However I can't find any way to crop a photo, can you help please?

**Reply by jpbro** [reply]
Select the area that you want to keep, then click the _Image_ menu > _Crop to selection_ option.

**Reply by zirrania** [reply]
Thank you very much. Easy when you know how!

**Reply by tannerhelland** [reply]
(Thank you for answering this so quickly, @jpbro !)

@zirrania - Hello!  I'm the developer of PhotoDemon.  Can I ask where you *first* looked for a way to crop an image?  I'm curious how I can make this feature more discoverable for new users.  Thank you!

**Comment by mfmdawdeh** [comment]
Welcome
I apologize
Which image has been cropped and what is required?
The program was deleted after the translation was completed
I am not a user but a software translator
________________________________
من: Tanner ***@***.***>
‏‏تم الإرسال: 15 تشرين الثاني, 2022 04:30 م
إلى: tannerhelland/PhotoDemon ***@***.***>
نسخة: mfmdawdeh ***@***.***>; Mention ***@***.***>
‏‏الموضوع: Re: [tannerhelland/PhotoDemon] Welcome to the PhotoDemon Forum! (Discussion #344)


(Thank you for answering this so quickly, @jpbro<https://github.com/jpbro> !)

@zirrania<https://github.com/zirrania> - Hello! I'm the developer of PhotoDemon. Can I ask where you first looked for a way to crop an image? I'm curious how I can make this feature more discoverable for new users. Thank you!

**Reply by tannerhelland** [reply]
Hello again, @mfmdawdeh !  You are replying to a generic discussion thread at PhotoDemon's GitHub forums.  I believe this is because you previously posted a comment at this link:

https://github.com/tannerhelland/PhotoDemon/discussions/344

I unfortunately did not receive your Arabic translation for PhotoDemon.  GitHub *does* allow users to post quick replies to discussion topics via email, but it does *not* allow attachments on these replies.  Attachments can only be uploaded at the discussion page.

I will happily add your translation to the project as soon as I receive it, but as mentioned in my previous reply, I need the XML file itself.  You can attach the XML translation file to any GitHub discussion page (or even better, create a new issue with it attached - https://github.com/tannerhelland/PhotoDemon/issues).  You can also email it to me directly at tanner`(at)`photodemon.org.

Thank you 🙏

**Comment by zirrania** [comment]
Well all other image editing programs I've seen have a small icon in the
top or side tools bar. I wasn't sure how to do it. I looked under "Edit"
and I tried selecting the shape of the crop size I wanted and then looking
in Edit.  I didn't think to look in "Image", though in hindsight I should
have tried that.  If you could put in a cop icon somewhere it would be a
lot easier to find, I think.

Hope that helped,
kind regards
Sue

**Reply by tannerhelland** [reply]
Thank you so much for the reply, Sue.  It helps a lot to hear what new users expect from the software.  I will modify my TODO list to prioritize a dedicated Crop tool.

**Comment by mfmdawdeh** [comment]
Hi.
Here is the Arabic XML file

________________________________
من: Tanner ***@***.***>
‏‏تم الإرسال: 16 تشرين الثاني, 2022 04:23 م
إلى: tannerhelland/PhotoDemon ***@***.***>
نسخة: mfmdawdeh ***@***.***>; Mention ***@***.***>
‏‏الموضوع: Re: [tannerhelland/PhotoDemon] Welcome to the PhotoDemon Forum! (Discussion #344)


Hello again, @mfmdawdeh<https://github.com/mfmdawdeh> ! You are replying to a generic discussion thread at PhotoDemon's GitHub forums. I believe this is because you previously posted a comment at this link:

#344<https://github.com/tannerhelland/PhotoDemon/discussions/344>

I unfortunately did not receive your Arabic translation for PhotoDemon. GitHub does allow users to post quick replies to discussion topics via email, but it does not allow attachments on these replies. Attachments can only be uploaded at the discussion page.

I will happily add your translation to the project as soon as I receive it, but as mentioned in my previous reply, I need the XML file itself. You can attach the XML translation file to any GitHub discussion page (or even better, create a new issue with it attached - https://github.com/tannerhelland/PhotoDemon/issues). You can also email it to me directly at tanner(at)photodemon.org.

Thank you 🙏

**Comment by CyrilTungstem** [comment]
Yo Tanner!

Photodemon looks neat but it's a shame that there's no 64bit version and version for linucks

**Comment by nishineineigeneinei1** [comment]
请问，左侧的工具栏是无法被修改的吗？
我常需要用复制和粘贴，但在9.0版本中右键无法呼出菜单，只能通过上方的 编辑 区找到，这有点麻烦，有什么解决办法吗？
Excuse me, is the toolbar on the Left unable to be modified? 
I often need to copy and paste, but in version 9.0 I cannot right-click to bring up the menu. It can only be found through the editing area above. This is a bit troublesome. Is there any solution?

**Reply by tannerhelland** [reply]
Thank you for your question, @nishineineigeneinei1 .  Right now, I'm afraid there is not a way to add new buttons to the left-hand toolbar.

Where would you like to right-click to raise a popup menu with "Copy" and "Paste" actions?  On the image itself?  Somewhere else?  I can possibly look at adding this behavior, but I don't really want to enable it for right-clicks on the main image.  (I was planning to use this for raising paint tool and color options in the future, like other photo editors.) 

Maybe I add a menu like this for right-clicks somewhere else on the main interface?  I am open to suggestions.
